src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -13,10 +13,24 @@ namespace System.Net.NetworkInformation
     public class NetworkChange
     {
         private static NetworkAddressChangedEventHandler s_addressChangedSubscribers;
+        private static NetworkAvailabilityChangedEventHandler s_availabilityChangedSubscribers;
         private static volatile int s_socket = 0;
         private static readonly object s_lockObj = new object();
+        private static readonly object s_availabilityLock = new object();
         private static readonly Interop.Sys.NetworkChangeEvent s_networkChangeCallback = ProcessEvent;
 
+        // The "leniency" window for NetworkAvailabilityChanged socket events.
+        // All socket events received within this duration will be coalesced into a
+        // single event. Generally, many route changed events are fired in succession,
+        // and we are not interested in all of them, just the fact that network availability
+        // has potentially changed as a result.
+        private const int AvailabilityTimerWindowMilliseconds = 150;
+        private static Timer s_availabilityTimer;
+        private static bool s_availabilityHasChanged;
+
+        // Introduced for supporting design-time loading of System.Windows.dll
+        public static void RegisterNetworkChange(NetworkChange nc) { }
```

Comment:
I suppose it wouldn't hurt. [Obsolete]'ing is really a compile-time thing, but it makes sense for both sides to agree.